docs/adr/000x-modelling-history.md

```diff
@@ -0,0 +1,250 @@
+historical concerns in the same data stores, thus ensuring history remains a
+first class citizen. A nice side effect of having immutable models is
+this opens options for caching. Since data for that
+model will never change it can effectively be cached forever.
+
+### Breakdown of Revision
+
+Revision is a model that stores little data itself and joins to other models
+that store the data. This can be visualised as:
+
+![Revision breakdown](000x/revision-diagram.png)
+
+The intention of breaking this up is to be conservative with the amount of data
+duplicated for both storage and maintenance reasons. Thus when a user edits
+the title of an edition a new ContentRevision object is created and the existing
+objects (TagsRevision, MetadataRevision and ImageRevisions) are associated with
+the next revision. An ImageRevision is modelled in a similar way to a Revision
+and this is explained further in [Image modelling](#image-modelling).
+
+It is intended that [delegation][delegate] be used when interfacing with a
+revision so that the caller need not be concerned with which sub-revision
+stores particular fields. This allows a revision to have a rich interface
+despite storing a low amount of data directly.
+
+### Image modelling
+
+Content Publisher supports a user uploading image files and referencing them
+in a revision of a document. They have metadata and editable properties that a
+user can change, of which a history is stored. A single image file uploaded
+produces multiple files that are uploaded to Asset Manager for different sizing
+variations. Images are modelled in a similar way to Revision with an
+Image::Revision model, as represented below:
+
+![Image Revision breakdown](000x/image-revision-diagram.png)
+
+The Image model itself is used for continuation between image revisions. It is
+known that two Image::Revisions are versions of the same item if they share the
+same Image association.
+
+The data of an Image::Revision is stored between an Image::FileRevision and an
+Image::MetadataRevision. These two objects differ by the fact that any change to
+Image::FileRevision requires changes to the resultant Asset Manager files (such
+as crop dimensions), whereas Image::MetadataRevision stores accompanying data
+that doesn't affect the Asset Manager files (such as alt text).
+
+Image::FileRevision is associated with an ActiveStorage::Blob object that is
+responsible for managing the storage of the source file. It also has a one to
+many association with Image::Asset, each of these represent resulting files that
+are uploaded to Asset Manager. The Image::Asset model stores the URL to the
+Asset Manager file and what state the file is on Asset Manager.
+
+### Timeline
+
+The TimelineEntry model represents an event that should be shown to a user as
+part of a visual timeline of a document's history. It stores the event which
+occurred and associations to relevant models. The sole purpose of this model
```

Comment:
Perhaps add 'polymorphic' here to indicate the association is very flexible.

Reply:
The reasoning is getting a bit lost here in trying to explain how things used to be. I think this change might help: 

'The sole purpose...' -> 'At the time of writing it wasn't yet determined what the timeline would show, and therefore it wasn't clear exactly how best to model them. In order to support future iteration on this model, it should not be used outside of the context of displaying a timeline'.

We could then modify the following paragraph to explain one of the consequences of this is/would-be to re-think the associations for Removals and Withdrawals.

Reply:
I've changed this around.

I've amended your suggestion though as that implies that we don't intend to use it for other reasons because we don't know the future iteration. Whereas that is only one of the reasons, the other being separation of concerns.